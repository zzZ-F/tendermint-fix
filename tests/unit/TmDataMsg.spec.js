import { mount } from "@vue/test-utils"
import htmlBeautify from "html-beautify"
import { TmDataMsg } from "@/index"

describe("TmDataMsg", () => {
  let wrapper

  let propsData = {
    title: "sOmEtHiNg Is HaPpEnIng!!!",
    subtitle: "Oh my! What could it be?",
    icon: "help",
    spin: false
  }

  beforeEach(() => {
    wrapper = mount(TmDataMsg, { propsData })
  })

  it("has a title from props", () => {
    expect(wrapper.vm.title).toBe("sOmEtHiNg Is HaPpEnIng!!!")
  })

  it("has the expected html structure", () => {
    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
  })

  it("has an icon", () => {
    expect(wrapper.find(".tm-data-msg__icon i.material-icons").text()).toBe(
      "help"
    )
  })

  it("doesn't have a spinning icon", () => {
    expect(
      wrapper.find(".tm-data-msg__icon").contains("i.material-icons.fa-spin")
    ).toBe(false)
  })

  it("has a subtitle", () => {
    expect(wrapper.find(".tm-data-msg__subtitle").text()).toBe(
      "Oh my! What could it be?"
    )
  })
})
