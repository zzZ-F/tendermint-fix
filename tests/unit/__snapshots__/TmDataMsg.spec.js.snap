// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmDataMsg has the expected html structure 1`] = `
<div class="tm-data-msg">
  <div class="tm-data-msg__icon">
    <i class="material-icons">help</i>
  </div>
  <div class="tm-data-msg__text">
    <div class="tm-data-msg__title">
      sOmEtHiNg Is HaPpEnIng!!!
    </div>
    <div class="tm-data-msg__subtitle">
      Oh my! What could it be?
    </div>
  </div>
</div>
`;
